include/triton/Tools/Sys/GetEnv.hpp

```diff
@@ -37,6 +37,7 @@ inline const std::set<std::string> ENV_VARS = {
     "DISABLE_LLVM_OPT",
     "DISABLE_MMA_V3",
     "DISABLE_PTXAS_OPT",
+    "ASM_ENABLE_DUMP",
```

Comment:
Unify this with `AMDGCN_ENABLE_DUMP`?

Reply:
I've considered this, although I don't want to break `AMDGCN_ENABLE_DUMP` for folks using it. Let me figure out what to do.

Reply:
@zhanglx13 would it be ok to replace `AMDGCN_ENABLE_DUMP` with `ASM_ENABLE_DUMP`?

Reply:
Actually I think I can drop `AMDGCN_ENABLE_DUMP` since it isn't used by `getBoolEnv` anywhere in Triton. It does seem to be used here however: https://github.com/openai/triton/blob/0b46687895f0bc7c4d5216150d8d5cfeb5b4e254/third_party/amd/backend/compiler.py#L222C12-L222C53

Reply:
It's only used by the amd backend. Let's confirm with Lixun that replacing it will not break any downstream tools.

Reply:
> @zhanglx13 would it be ok to replace AMDGCN_ENABLE_DUMP with ASM_ENABLE_DUMP?

@zhanglx13  @htyu  I've updated this in a way as to not break AMDGCN_ENABLE_DUMP immediately. Let me know what you think.